src/storage/v3/shard_rsm.cpp

```diff
@@ -0,0 +1,443 @@
+  std::map<PropertyId, Value> ret;
+
+  for (const auto &prop : props.value()) {
+    auto result = acc.GetProperty(prop, view);
+    if (result.HasError()) {
+      spdlog::debug("Encountered an Error while trying to get a vertex property.");
+      continue;
+    }
+    if (result.HasValue()) {
+      if (result.GetValue().IsNull()) {
+        spdlog::debug("The specified property does not exist but it should");
+        continue;
+      }
+
+      ret.emplace(prop, ToValue(result.GetValue()));
+    }
+  }
+
+  return ret;
+}
+
+std::optional<std::map<PropertyId, Value>> CollectAllPropertiesFromAccessor(
+    const memgraph::storage::v3::VertexAccessor &acc, memgraph::storage::v3::View view) {
+  std::map<PropertyId, Value> ret;
+  auto iter = acc.Properties(view);
+  if (iter.HasError()) {
+    spdlog::debug("Encountered an Error while trying to get vertex properties.");
+  }
+
+  for (const auto &[prop_key, prop_val] : iter.GetValue()) {
+    ret.emplace(prop_key, ToValue(prop_val));
+  }
+
+  return ret;
+}
+
+Value ConstructValueVertex(const memgraph::storage::v3::VertexAccessor &acc, memgraph::storage::v3::View view) {
+  // Get the vertex id
+  auto prim_label = acc.PrimaryLabel(view).GetValue();
+  Label value_label = {.id = prim_label};
+
+  auto prim_key = ConvertValueVector(acc.PrimaryKey(view).GetValue());
+  VertexId vertex_id = std::make_pair(value_label, prim_key);
+
+  // Get the labels
+  auto vertex_labels = acc.Labels(view).GetValue();
+  std::vector<Label> value_labels;
+  for (const auto &label : vertex_labels) {
+    Label l = {.id = label};
+    value_labels.push_back(l);
+  }
+
+  return Value({.id = vertex_id, .labels = value_labels});
+}
+
+}  // namespace
+
+namespace memgraph::storage::v3 {
+
+WriteResponses ShardRsm::ApplyWrite(CreateVerticesRequest &&req) {
+  auto acc = shard_->Access();
+
+  // Workaround untill we have access to CreateVertexAndValidate()
+  // with the new signature that does not require the primary label.
+  auto prim_label = acc.GetPrimaryLabel();
+
+  bool action_successful = true;
+
+  for (auto &new_vertex : req.new_vertices) {
+    if (!action_successful) {
+      break;
+    }
+    /// TODO(gvolfing) Remove this. In the new implementation each shard
+    /// should have a predetermined primary label, so there is no point in
+    /// specifying it in the accessor functions. Their signature will
+    /// change.
+    /// TODO(gvolfing) Consider other methods than converting. Change either
+    /// the way that the property map is stored in the messages, or the
+    /// signature of CreateVertexAndValidate.
+    auto converted_property_map = ConvertPropertyMap(new_vertex.properties);
+
+    // TODO(gvolfing) make sure you don't create vectors bigger than they actually should be, e.g.
+    // std::vector<memgraph::storage::v3::LabelId> converted_label_ids(new_vertex.label_ids.size());
```

Comment:
Is this comment still relevant? I think `reserve` + `emplace_back` is better than creating an explicitly sized `vector` with default value and then overriding them.